lib/Models/Catalog/Ows/WebMapServiceCatalogItem.ts

```diff
@@ -619,6 +623,104 @@ class WebMapServiceCatalogItem
     }
   );
 
+  fetchPalettes(): Promise<string[]> {
+    return new Promise((resolve, reject) => {
+      if (!this.isThredds) {
+        resolve([]);
+      }
+
+      if (this.palettes.length > 0) {
+        resolve([...this.palettes]);
+      }
+
+      const extractedUrl = this.uri!.clone()
+        .setSearch({
+          service: "WMS",
+          version: this.useWmsVersion130 ? "1.3.0" : "1.1.1",
+          request: "GetMetadata",
+          item: "layerDetails",
+          layerName: this.layersArray[0]
+        })
+        .toString();
+
+      if (!extractedUrl) {
+        resolve([]);
+      }
+      const proxiedUrl = proxyCatalogItemUrl(this, extractedUrl!);
+
+      fetch(proxiedUrl, { method: "GET" })
+        .then((response) => response.json())
+        .then((data) => {
+          this.setTrait(CommonStrata.user, "palettes", data.palettes);
+          this.setTrait(
+            CommonStrata.user,
+            "colorScaleMinimum",
+            data.scaleRange[0]
+          );
+          this.setTrait(
+            CommonStrata.user,
+            "colorScaleMaximum",
+            data.scaleRange[1]
+          );
+          this.setTrait(
+            CommonStrata.user,
+            "defaultPalette",
+            data.defaultPalette
+          );
+          this.setTrait(
+            CommonStrata.user,
+            "noPaletteStyles",
+            data.noPaletteStyles
+          );
+          this.setTrait(CommonStrata.user, "showPalettes", true);
+          resolve(data.palettes);
+        })
+        .catch((error) => {
+          reject(error);
+        });
+    });
+  }
+
+  @computed
+  get paletteDimensions(): SelectableDimensionEnum[] {
+    if (!this.isNcWMS || !this.showPalettes) {
+      return [];
+    }
+
+    const options: { name: string; id: string }[] = [];
+
+    this.palettes.map((palette) => {
+      options.push({
+        name: palette,
+        id: palette
+      });
+    });
+
+    return [
+      {
+        name: "Palettes",
+        id: `${this.uniqueId}-palettes`,
+        options,
+        selectedId: this.palette,
+        setDimensionValue: (
+          stratumId: string,
+          newPalette: string | undefined
+        ) => {
+          //if (this.stylesArray[0]) {
+          runInAction(() => {
+            this.setTrait(stratumId, "palette", newPalette);
+            const currentStyle = this.stylesArray[0]
+              ? this.stylesArray[0]
+              : "default";
+            const newStyles = currentStyle.split("/")[0] + "/" + newPalette;
+            this.setTrait(stratumId, "styles", newStyles);
```

Comment:
I think there are some issues with updating `styles` like this - you should keep `palette` and `styles` separate.

With the current implementation, `palette` is not used unless the user makes changes - for example - if `palette` was defined in a catalog JSON file, it wouldn't be used.

You can apply the palette to the `styles` parameter here instead. You should only apply the palette to the first style - as this maps to the first layers style.

https://github.com/TerriaJS/terriajs/blob/bacefe0a8ac9e69aceff4138f4eb4ce2b0714c9e/lib/Models/Catalog/Ows/WebMapServiceCatalogItem.ts#L566